src/System.Console/src/System/Console.cs

```diff
@@ -38,10 +43,65 @@ public static TextReader In
         {
             get
             {
-                return Volatile.Read(ref _in) ?? EnsureInitialized(ref _in, () =>
+                return Volatile.Read(ref s_in) ?? EnsureInitialized(ref s_in, () => ConsolePal.GetOrCreateReader());
```

Comment:
Were you going to make s_in, s_out, etc. volatile rather than using Volatile.* on every access?